src/utils/__tests__/parseJsDoc-test.js

```diff
@@ -64,6 +64,21 @@ describe('parseJsDoc', () => {
       });
     });
 
+    it('extracts jsdoc union type param', () => {
+      const docblock = `
+        @param {string|Object} bar
+      `;
+      expect(parseJsDoc(docblock)).toEqual({
+        description: null,
+        returns: null,
+        params: [{
+          name: 'bar',
+          type: {name: 'string|Object'},
```

Comment:
I agree we should keep this as an object since it's easier to use after. We should try to have a similar output object as the flow type definitions.
```
/**
 * @param test Test
 */
function (test: string | Object) {}
```
and
```
/**
 * @param {string | Object} test Test
 */
function (test) {}
```
should result in the same thing.

You could take a look at what the object we return for flow types look like and return something similar.